core/src/test/scala/unit/kafka/utils/JsonTest.scala

```diff
@@ -42,6 +44,28 @@ class JsonTest {
     val arrayNode = new ArrayNode(jnf)
     Vector(1, 2, 3).map(new IntNode(_)).foreach(arrayNode.add)
     assertEquals(Json.parseFull("[1, 2, 3]"), Some(JsonValue(arrayNode)))
+
+    // Test with encoder that properly escapes backslash and quotes
+    val map = Map("foo1" -> "bar1\\,bar2", "foo2" -> "\\bar")
+    val encoded = Json.encode(map)
+    val decoded = Json.parseFull(encoded)
+    assertEquals(Json.parseFull("""{"foo1":"bar1\\,bar2", "foo2":"\\bar"}"""), decoded)
+
+    // Test strings with non-escaped backslash and quotes. This is to verify that ACLs
+    // containing non-escaped chars persisted using 1.0 can be parsed.
+    def encodeWithoutEscaping(obj: Any): String = {
+      obj match {
+        case s: String => "\"" + s + "\""
+        case m: Map[_, _] => "{" +
+          m.map {
+            case (k, v) => encodeWithoutEscaping(k) + ":" + encodeWithoutEscaping(v)
+            case elem => throw new IllegalArgumentException(s"Invalid map element '$elem' in $obj")
+          }.mkString(",") + "}"
+        case o => Json.encode(o)
+      }
+    }
```

Comment:
Simple this is a reasonably simple string, can we not just provide the full string without encoding instead of implementing this method?